src/backend/distributed/commands/extension.c

```diff
@@ -78,40 +76,804 @@ ErrorIfUnstableCreateOrAlterExtensionStmt(Node *parsetree)
+
+		return pstrdup(newVersion);
+	}
+	else
+	{
+		return NULL;
+	}
+}
+
+
+/*
+ * PlanCreateExtensionStmt is called during the creation of an extension.
+ * It is executed before the statement is applied locally.
+ * We decide if the extension needs to be replicated to the worker, and
+ * if that is the case return a list of DDLJob's that describe how and
+ * where the extension needs to be created.
+ */
+List *
+PlanCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const char *queryString)
+{
+	List *commands = NIL;
+	const char *createExtensionStmtSql = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return NIL;
+	}
+
+	/*
+	 * If the extension command is a part of a bigger multi-statement transaction,
+	 * do not propagate it
+	 */
+	if (IsMultiStatementTransaction())
+	{
+		return NIL;
+	}
+
+	/* extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	/*
+	 * Make sure that no new nodes are added after this point until the end of the
+	 * transaction by taking a RowShareLock on pg_dist_node, which conflicts with the
+	 * ExclusiveLock taken by master_add_node.
+	 * This guarantees that all active nodes will have the extension, because they will
+	 * either get it now, or get it in master_add_node after this transaction finishes and
+	 * the pg_dist_object record becomes visible.
+	 */
+	LockRelationOid(DistNodeRelationId(), RowShareLock);
+
+	/*
+	 * Make sure that the current transaction is already in sequential mode,
+	 * or can still safely be put in sequential mode
+	 */
+	EnsureSequentialModeForExtensionDDL();
+
+	/*
+	 * Here we append "schema" field to the "options" list (if not specified)
+	 * to satisfy the schema consistency between worker nodes and the coordinator.
+	 */
+	AddSchemaFieldIfMissing(createExtensionStmt);
+
+	createExtensionStmtSql = DeparseTreeNode((Node *) createExtensionStmt);
+
+	/*
+	 * To prevent recursive propagation in mx architecture, we disable ddl
+	 * propagation before sending the command to workers.
+	 */
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) createExtensionStmtSql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * AddSchemaFieldIfMissing adds DefElem item for "schema" (if not specified
+ * in statement) to "options" list before deparsing the statement to satisfy
+ * the schema consistency between worker nodes and the coordinator.
+ */
+static void
+AddSchemaFieldIfMissing(CreateExtensionStmt *createExtensionStmt)
+{
+	List *optionsList = createExtensionStmt->options;
+
+	Value *schemaNameValue = GetExtensionOption(optionsList, "schema");
+
+	if (!schemaNameValue)
+	{
+		char *currentSchemaName = GetCurrentSchema();
+
+		Node *schemaNameArg = (Node *) makeString(currentSchemaName);
+
+		/* set location to -1 as it is unknown */
+		int location = -1;
+
+		DefElem *newDefElement = makeDefElem("schema", schemaNameArg, location);
```

Comment:
Does this work correctly for extensions with a schema defined in their control file (like citus)?

Reply:
Actually, I didn't test my code with such an extension. Maybe I should test such an extension

Reply:
Actually, I didn't test my code with such an extension. Maybe I should test such an extension

Reply:
Now I tried it on hll and it worked

Reply:
Verified that creating Citus extension works fine (by changing some code). Also, other extensions as well